### PR TITLE
test: refactor the existing KV RESTful test cases and also add new cases

### PR DESCRIPTION
**Pull request by ahrtr**
All RESTful API in KV service are covered now.

part of https://github.com/etcd-io/etcd/issues/16507

also link to https://github.com/etcd-io/etcd/pull/16454

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: tests/e2e/v3_curl_kv_test.go [hunk 1]
```diff
@@ -118,18 +153,30 @@ func testV3CurlTxn(cx ctlCtx) {
...
-		cx.t.Fatal(jerr)
-	}
+	require.NoError(cx.t, jerr)
+
 	expected := `"succeeded":true,"responses":[{"response_put":{"header":{"revision":"2"}}}]`
-	p := cx.apiPrefix
-	if err := e2e.CURLPost(cx.epc, e2e.CURLReq{Endpoint: path.Join(p, "/kv/txn"), Value: string(jsonDat), Expected: expect.ExpectedResponse{Value: expected}}); err != nil {
-		cx.t.Fatalf("failed testV3CurlTxn txn with curl using prefix (%s) (%v)", p, err)
-	}
+	err := e2e.CURLPost(cx.epc, e2e.CURLReq{Endpoint: "/v3/kv/txn", Value: string(jsonDat), Expected: expect.ExpectedResponse{Value: expected}})
```
Comment (fuweid) [inline tests/e2e/v3_curl_kv_test.go:159]: it could be easy to read if using multiple lines here.

**Review by fuweid — APPROVED** [review]
LGTM

File: tests/e2e/v3_curl_kv_test.go (hunk 1 above)
Comment (ahrtr) [reply tests/e2e/v3_curl_kv_test.go]: updated.

**Review by fuweid — APPROVED** [review]
LGTM, Thanks!

**Comment by ahrtr** [comment]
thx @fuweid 

cc @jmhbnz @serathius @wenjiaswe PTAL

**Review by jmhbnz — COMMENTED** [review]
Is the intent we also refactor and rename other tests for example in `tests/e2e/v3_curl_test.go` or `tests/e2e/v3_curl_maintenance_test.go`.

The changes in this file are the only instances of the new naming so just want to check if we want to stay aligned on old naming or gradually rename all of them?

**Comment by ahrtr** [comment]
> Is the intent we also refactor and rename other tests for example in `tests/e2e/v3_curl_test.go` or `tests/e2e/v3_curl_maintenance_test.go`?

Yes, let's do it in separate PRs.

**Review by jmhbnz — APPROVED** [review]
Code changes LGTM - Just a question about direction for naming tests.

**Comment by ahrtr** [comment]
> > Is the intent we also refactor and rename other tests for example in `tests/e2e/v3_curl_test.go` or `tests/e2e/v3_curl_maintenance_test.go`?
> 
> Yes, let's do it in separate PRs.

Also pls feel free to raise PR to do it if you want.

File: tests/e2e/v3_curl_kv_test.go [hunk 2]
```diff
@@ -16,86 +16,121 @@ package e2e
...
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			testCtl(t, f, tc.cfg)
+		})
 	}
 }
 
-func testV3CurlPutGet(cx ctlCtx) {
+func testCurlV3KVBasicOperation(cx ctlCtx) {
```
Comment (serathius) [inline tests/e2e/v3_curl_kv_test.go:76]: Using `ctlCtx` is the old and at least for me a deprecated way to write e2e test. Would appreciate if you considered a followup PR to refactor it.

**Review by serathius — APPROVED** [review]

**Pull request merged**